charts/vsphere-cpi/v1.0.0/values.yaml

```diff
@@ -1,11 +1,31 @@
-vCenter:
+cloudConfig:
   host: ""
-  insecureflag: "1"
-  port: "443"
+  insecureFlag: true
+  port: 443
   datacenters: ""
+  template: |
+    [Global]
+    secret-name = {{ .Values.secret.name | quote }}
+    secret-namespace = {{ .Release.Namespace | quote }}
+    port = {{ .Values.cloudConfig.port | quote }}
+    insecure-flag = {{ .Values.cloudConfig.insecureFlag | quote }}
+
+    [VirtualCenter {{ .Values.cloudConfig.host | quote }}]
+    datacenters = {{ .Values.cloudConfig.datacenters | quote }}
+
+secret:
```

Comment:
Why is secret nested in the CSI chart under csiController, but not nested under CloudConfig here?

Reply:
THis is how it's shown in CSI, so we should be consistent:
```
csiController:  
    username: ""
    password: ""
    secret:
      name: "vsphere-config-secret"
      generate: true
```

Reply:
CPI uses a config map (cloud-config) that references a secret with credentials, whereas CSI doesn't use a configMap. Instead, it has the contents of `vsphere-csi.conf` which includes vCenter info and credentials, all inside a secret.
I know is not ideal, so I tried to use a config map for CSI and reference a secret as well, but it doesn't work because that's how they have it upstream.